gdynet/preprocess.py

```diff
@@ -169,6 +169,33 @@ def construct_graph(self, traj_coords, lattices, atom_types, target_index):
                     'target_index': target_index,
                     'nbr_lists': nbr_lists,
                     'nbr_dists': nbr_dists}
+        elif self.backend == 'ndirect':
+            stcs = [Structure(lattice=lattices[i],
```

Comment:
Don’t use such complex list comprehensions. Use a for loop for code readability.